src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForFluentDiagnosticAnalyzer.cs

```diff
@@ -224,21 +235,44 @@ private static bool AnalyzeInvocation(
             {
                 if (objectCreation is not
                     {
-                        ArgumentList: null or { Arguments.Count: 0 },
                         Initializer: null or { RawKind: (int)SyntaxKind.CollectionInitializerExpression }
                     })
                 {
                     return false;
                 }
 
+                existingInitializer = objectCreation.Initializer;
+
                 if (!IsLegalInitializer(objectCreation.Initializer))
                     return false;
 
                 if (!IsListLike(current))
                     return false;
 
-                existingInitializer = objectCreation.Initializer;
-                return true;
+                if (objectCreation.ArgumentList is { Arguments.Count: 1 })
+                {
+                    // Can take a single argument if that argument is itself a collection.
+                    var argumentType = semanticModel.GetTypeInfo(objectCreation.ArgumentList.Arguments[0].Expression, cancellationToken).Type;
+                    if (argumentType is null)
+                        return false;
+
+                    if (!Equals(argumentType.OriginalDefinition, ienumerableOfTType) &&
+                        !argumentType.AllInterfaces.Any(i => Equals(i.OriginalDefinition, ienumerableOfTType)))
+                    {
+                        return false;
+                    }
+
+                    // Add the arguments to the pre-matches.  They will execute before the initializer values are added.
+                    AddArgumentsInReverse(preMatchesInReverse, objectCreation.ArgumentList.Arguments, useSpread: true);
```

Comment:
new logic.  if the user wrote `new List<...>(someValuesToCopy)` we'll create a collection expr starting with `[.. someValuesToCopy`.  Any existing initializer values will be added after this.  And any more fluent additions will go after that.